truffle/src/main/ruby/core/rubinius/api/shims/tuple.rb

```diff
@@ -28,6 +22,48 @@ def copy_from(other, start, length, dest)
 
 class Array
 
+  def copy_from(other, start, length, dest)
+    # Rubinius.primitive :tuple_copy_from
```

Comment:
@nirvdrum where are these primitives added? I'll add the null case.

Working with the array methods so far, I think there be a more general purpose primitive that we could create that could replace `copy_from`. `copy_from` for example doesn't allow growing or shrinking the array.

I think it would be something like this but in java:
```ruby
def splice_arrays( array, other, a, b, m, n)
  array[a,n] = other[m,n]
end
```